tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2260,14 +2310,19 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
               "Invalid tree in JSDesugar.transformExpr() of class " +
               tree.getClass)
       }
+
+      if (preserveChar || tree.tpe != CharType)
+        baseResult
+      else
+        genCallHelper("bC", baseResult)
```

Comment:
Uh oh, indeed. The problem is not here, though. It's only that `Block`/`If`/etc. should never just forward `preserveChar`. They must use their `tpe` as expected type for the embedded expressions.

It's weird that nothing in the CI caught this. Probably because it's very hard or impossible to write user code that results in some IR that would expose this issue. I'll try to craft a failing test case.